File: solutions/BridgeBuildingDiv2/solve/javascript/BridgeBuildingDiv2Solver.js
#! /usr/bin/env node

var fs = require("fs");
var path = require("path");
var tc = null;

function init() {
    var gettcHome = process.env.GETTC_HOME || path.join(process.env.HOME, ".gettc");
    var includeDir = path.join(gettcHome, "include", "javascript");
    tc = require(path.join(includeDir, "topcoder"));
}

function main() { try {
    var input = fs.readFileSync(process.argv[2], { encoding: "ascii" });
    var reader = new tc.Reader(input);
    var a = reader.next("int[]"); reader.next();
    var b = reader.next("int[]"); reader.next();
    var K = reader.next("int");

    var BridgeBuildingDiv2 = require("./BridgeBuildingDiv2");
    var result = BridgeBuildingDiv2.minDiameter(a, b, K);
    
    var writer = new tc.Writer();
    writer.next(result, "int");
    fs.writeFileSync(process.argv[3], writer.toString(), { encoding: "ascii" });
} catch (err) {
    console.log(err.toString());
    console.trace();
}}

init();
main();

File: solutions/BridgeBuildingDiv2/solve/ruby/BridgeBuildingDiv2Solver.rb
#! /usr/bin/env ruby

require_relative "BridgeBuildingDiv2"

require "gettc/types"
include Gettc

def init()
    gettc_home = File.join(ENV["HOME"], ".gettc")
    if ENV.has_key?("GETTC_HOME")
        gettc_home = ENV["GETTC_HOME"]
    end 
    gettc_home = File.expand_path(gettc_home)
    $LOAD_PATH << File.join(gettc_home, "include", "ruby")
    require "topcoder"
    include TopCoder
end

def main()
    reader = Reader.new(IO.read(ARGV[0]))
    a = reader.next(TArray.new(TInt))
    reader.next()
    b = reader.next(TArray.new(TInt))
    reader.next()
    K = reader.next(TInt)

    result = BridgeBuildingDiv2.new().minDiameter(a, b, K)
    IO.write(ARGV[1], Writer.new().next(result, TInt).to_s())
end
init()
main()
